Index: kdepim/kpilot/conduits/todo/todoconduit.cc
===================================================================
--- kdepim/kpilot/conduits/todo/todoconduit.cc	(revision 915906)
+++ kdepim/kpilot/conduits/todo/todoconduit.cc	(revision 915907)
@@ -68,6 +68,7 @@
 	
 	TodoSettings::self()->readConfig();
 	d->fCollectionId = TodoSettings::akonadiCollection();
+	d->fPrevCollectionId = TodoSettings::prevAkonadiCollection();
 }
 
 bool TodoConduit::initDataProxies()
