Index: kdepim/kpilot/conduits/akonadibase/akonadirecord.cc
===================================================================
--- kdepim/kpilot/conduits/akonadibase/akonadirecord.cc	(revision 915830)
+++ kdepim/kpilot/conduits/akonadibase/akonadirecord.cc	(revision 915831)
@@ -47,12 +47,14 @@
 	d->fItem = item;
 	d->fLastSyncDateTime = lastSync.toUTC(); // Item times are saved in UTC time.
 	d->fDeleted = false;
+	d->fDummy = false;
 }
 
 AkonadiRecord::AkonadiRecord( const QString& id ) : d( new AkonadiRecordPrivate )
 {
 	d->fTempId = id;
 	d->fDeleted = true;
+	d->fDummy = false;
 }
 
 AkonadiRecord::~AkonadiRecord()
