--- a/akonadi/agents/nepomuk_contact_feeder/CMakeLists.txt
+++ b/akonadi/agents/nepomuk_contact_feeder/CMakeLists.txt
@@ -5,7 +5,6 @@
   ${NEPOMUK_INCLUDES}
   )
 
-if(HAVE_NIE)
   set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${KDE4_ENABLE_EXCEPTIONS}" )
 
   kde4_add_executable(akonadi_nepomuk_contact_feeder RUN_UNINSTALLED nepomukcontactfeeder.cpp)
@@ -22,4 +21,3 @@
   install(TARGETS akonadi_nepomuk_contact_feeder ${INSTALL_TARGETS_DEFAULT_ARGS})
   install(FILES nepomukcontactfeeder.desktop DESTINATION "${CMAKE_INSTALL_PREFIX}/share/akonadi/agents")
 
-endif(HAVE_NIE)
--- a/akonadi/agents/nepomuk_email_feeder/CMakeLists.txt
+++ b/akonadi/agents/nepomuk_email_feeder/CMakeLists.txt
@@ -4,7 +4,6 @@
   ${Boost_INCLUDE_DIR}
   )
 
-if(HAVE_NIE)
   set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${KDE4_ENABLE_EXCEPTIONS}" )
 
   set(akonadi_nepomuk_email_feeder_SRCS nepomukemailfeeder.cpp)
@@ -26,4 +25,3 @@
 
   install(TARGETS akonadi_nepomuk_email_feeder ${INSTALL_TARGETS_DEFAULT_ARGS})
   install(FILES nepomukemailfeeder.desktop DESTINATION "${CMAKE_INSTALL_PREFIX}/share/akonadi/agents")
-endif(HAVE_NIE)
