--- a/akonadi/kabc/CMakeLists.txt
+++ b/akonadi/kabc/CMakeLists.txt
@@ -20,7 +20,7 @@
 qt4_wrap_ui(kabcakonadi_LIB_SRC kabcitemeditor.ui)
 kde4_add_library( akonadi-kabccommon SHARED ${kabcakonadi_LIB_SRC} )
 
-target_link_libraries( akonadi-kabccommon ${KDE4_AKONADI_LIBS} ${KDE4_KABC_LIBS} )
+target_link_libraries( akonadi-kabccommon ${KDE4_AKONADI_LIBS} ${KDE4_KABC_LIBS} ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${QT_QTGUI_LIBRARY} )
 set_target_properties( akonadi-kabccommon PROPERTIES VERSION ${GENERIC_LIB_VERSION} 
                                                SOVERSION ${GENERIC_LIB_SOVERSION} 
                                                ${KDE4_DISABLE_PROPERTY_}LINK_INTERFACE_LIBRARIES "${KDE4_KABC_LIBRARY}" )
--- a/akonadi/kcal/CMakeLists.txt
+++ b/akonadi/kcal/CMakeLists.txt
@@ -18,6 +18,6 @@
 
 kde4_add_library( akonadi-kcal SHARED ${kcalakonadi_LIB_SRC} )
 
-target_link_libraries( akonadi-kcal ${KDE4_AKONADI_LIBS} ${KDE4_KCAL_LIBS} )
+target_link_libraries( akonadi-kcal ${KDE4_AKONADI_LIBS} ${KDE4_KCAL_LIBS} ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${QT_QTGUI_LIBRARY} )
 set_target_properties( akonadi-kcal PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
 install(TARGETS akonadi-kcal ${INSTALL_TARGETS_DEFAULT_ARGS})
--- a/akonadi/opensync/CMakeLists.txt
+++ b/akonadi/opensync/CMakeLists.txt
@@ -23,6 +23,7 @@
   ${KDE4_KDECORE_LIBS}
   ${KDE4_AKONADI_LIBS}
   ${KDE4_KCAL_LIBS}
+  ${KDE4_KABC_LIBS}
 )
 
 OPENSYNC_PLUGIN_INSTALL( akonadi_opensync_plugin )
--- a/kpilot/conduits/contacts/CMakeLists.txt
+++ b/kpilot/conduits/contacts/CMakeLists.txt
@@ -23,7 +23,7 @@
 kde4_add_library( kpilot_conduit_contacts SHARED ${contacts_SRCS} )
 
 target_link_libraries( kpilot_conduit_contacts
-	${KDE4_KCAL_LIBS} ${KDE4_AKONADI_LIBS}
+	${KDE4_KCAL_LIBS} ${KDE4_AKONADI_LIBS} ${KDE4_KABC_LIBS}
 	kpilot_akonadibase kpilot_conduit_base kpilot
 )
 
--- a/kresources/akonadi/kabc/CMakeLists.txt
+++ b/kresources/akonadi/kabc/CMakeLists.txt
@@ -8,7 +8,7 @@
 
 kde4_add_plugin(kabc_akonadi ${kabc_akonadi_SRCS})
 
-target_link_libraries(kabc_akonadi ${KDE4_KABC_LIBS} ${KDE4_AKONADI_LIBS} ${KDE4_KUTILS_LIBS})
+target_link_libraries(kabc_akonadi ${KDE4_KABC_LIBS} ${KDE4_AKONADI_LIBS} ${KDE4_KUTILS_LIBS} ${KDE4_KRESOURCES_LIBS})
 
 
 install(TARGETS kabc_akonadi  DESTINATION ${PLUGIN_INSTALL_DIR} )
--- a/kresources/akonadi/kcal/CMakeLists.txt
+++ b/kresources/akonadi/kcal/CMakeLists.txt
@@ -9,7 +9,7 @@
 
 kde4_add_plugin(kcal_akonadi ${kcal_akonadi_SRCS})
 
-target_link_libraries(kcal_akonadi ${KDE4_KCAL_LIBS} ${KDE4_AKONADI_LIBS} akonadi-kcal)
+target_link_libraries(kcal_akonadi ${KDE4_KCAL_LIBS} ${KDE4_AKONADI_LIBS} ${KDE4_KRESOURCES_LIBS} ${KDE4_KABC_LIBS} ${KDE4_KUTILS_LIBS} akonadi-kcal)
 
 
 install(TARGETS kcal_akonadi  DESTINATION ${PLUGIN_INSTALL_DIR} )
--- a/kresources/groupwise/soap/CMakeLists.txt
+++ b/kresources/groupwise/soap/CMakeLists.txt
@@ -26,7 +26,7 @@
    )
 
 kde4_add_library(gwsoap SHARED ${gwsoap_SRCS})
-target_link_libraries(gwsoap ${KDE4_KCAL_LIBS} ${KDE4_KABC_LIBS})
+target_link_libraries(gwsoap ${KDE4_KCAL_LIBS} ${KDE4_KABC_LIBS} ${KDE4_KDECORE_LIBS} ${QT_QTDBUS_LIBRARY})
 set_target_properties(gwsoap PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
 install(TARGETS gwsoap ${INSTALL_TARGETS_DEFAULT_ARGS})
 
--- a/strigi-analyzer/vcf/CMakeLists.txt
+++ b/strigi-analyzer/vcf/CMakeLists.txt
@@ -6,6 +6,6 @@
 
 kde4_add_library(vcf MODULE ${vcfendanalyzer_SRCS})
 
-target_link_libraries(vcf ${STRIGI_STREAMANALYZER_LIBRARY} ${KDE4_KABC_LIBS})
+target_link_libraries(vcf ${STRIGI_STREAMANALYZER_LIBRARY} ${KDE4_KABC_LIBS} ${KDE4_KDECORE_LIBS})
 set_target_properties(vcf PROPERTIES PREFIX strigiea_)
 install(TARGETS vcf LIBRARY DESTINATION ${LIB_INSTALL_DIR}/strigi)
