--- a/libkdepim/CMakeLists.txt
+++ b/libkdepim/CMakeLists.txt
@@ -91,7 +91,7 @@
 
 kde4_add_library(kdepim SHARED ${kdepim_LIB_SRCS})
 
-target_link_libraries(kdepim  ${KDE4_KDE3SUPPORT_LIBS} ${QT_QTDESIGNER_LIBRARY} ${KDE4_KRESOURCES_LIBS} ${KDE4_KABC_LIBS} ${KDE4_KCAL_LIBS}  kimproxy ${KDE4_KPIMUTILS_LIBS} ${KDE4_KLDAP_LIBS} ${KDE4_KPIMIDENTITIES_LIBS})
+target_link_libraries(kdepim  ${KDE4_KDE3SUPPORT_LIBS} ${QT_QTDESIGNER_LIBRARY} ${KDE4_KRESOURCES_LIBS} ${KDE4_KABC_LIBS} ${KDE4_KCAL_LIBS}  kimproxy ${KDE4_KPIMUTILS_LIBS} ${KDE4_KLDAP_LIBS} ${KDE4_KPIMIDENTITIES_LIBS} ${KDE4_KPARTS_LIBRARY} ${QT_QTDBUS_LIBRARY} ${QT_QTXML_LIBRARY})
 
 if(MINGW)
 target_link_libraries(kdepim oleaut32)
--- a/akregator/src/CMakeLists.txt
+++ b/akregator/src/CMakeLists.txt
@@ -31,7 +31,7 @@
 
 kde4_add_library(akregatorprivate SHARED ${akregatorprivate_LIB_SRCS})
 
-target_link_libraries(akregatorprivate kdepim  ${KDE4_KHTML_LIBS} ${KDE4_SYNDICATION_LIBS} akregatorinterfaces )
+target_link_libraries(akregatorprivate kdepim  ${KDE4_KHTML_LIBS} ${KDE4_SYNDICATION_LIBS} ${KDE4_SOLID_LIBRARY} akregatorinterfaces )
 
 set_target_properties(akregatorprivate PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
 install(TARGETS akregatorprivate DESTINATION ${LIB_INSTALL_DIR})
--- a/ktimetracker/CMakeLists.txt
+++ b/ktimetracker/CMakeLists.txt
@@ -74,7 +74,7 @@
 
 kde4_add_plugin(karmpart ${karmpart_PART_SRCS})
 
-target_link_libraries(karmpart  ${KDE4_KCAL_LIBS}  ${KDE4_KIO_LIBS} kdepim kcal_resourceremote)
+target_link_libraries(karmpart  ${KDE4_KCAL_LIBS}  ${KDE4_KIO_LIBS} ${X11_X11_LIB} kdepim kcal_resourceremote)
 if(X11_Xscreensaver_LIB)
 	target_link_libraries(karmpart  ${X11_Xscreensaver_LIB} )
 endif(X11_Xscreensaver_LIB)
--- a/korganizer/CMakeLists.txt
+++ b/korganizer/CMakeLists.txt
@@ -226,7 +226,8 @@
 
 add_dependencies(korganizerprivate kmail)
 
-target_link_libraries(korganizerprivate ${KDE4_KNEWSTUFF2_LIBS} ${KDE4_KABC_LIBRARY} ${KDE4_KUTILS_LIBRARY} ${KDE4_KCAL_LIBRARY} ${KDE4_KPIMIDENTITIES_LIBRARY} ${KDE4_KMIME_LIBRARY}
+target_link_libraries(korganizerprivate ${KDE4_KNEWSTUFF2_LIBS} ${KDE4_KABC_LIBRARY} ${KDE4_KUTILS_LIBRARY} ${KDE4_KCAL_LIBRARY} ${KDE4_KPIMIDENTITIES_LIBRARY} ${KDE4_KMIME_LIBRARY} ${ZLIB_LIBRARY}
+
                                                       kdepim korganizer_eventviewer korganizer_calendar kocorehelper
                                                       korg_stdprinting kholidays korganizer_interfaces )
 
