From: Christophe Giboudeaux <cgiboudeaux@gmail.com>
Subject: Make sure that the archive exists before calling markAsFetchNow.
Origin: backport http://websvn.kde.org/?revision=1208110&view=revision
Bug: https://bugs.kde.org/show_bug.cgi?id=254323
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=605094
Forwarded: yes
Applied-Upstream: 4.4.9
Last-Update: 2011-01-20

Patch by Guy Shapiro. Thank you.

--- a/akregator/src/feed.cpp
+++ b/akregator/src/feed.cpp
@@ -682,6 +682,7 @@ void Feed::fetchCompleted(Syndication::L
 
 void Feed::markAsFetchedNow()
 {
+    if ( d->archive )
     d->archive->setLastFetch( QDateTime::currentDateTime().toTime_t());
 }
 
